front/src/pages/AddItem/AddItem.tsx

```diff
@@ -1,28 +1,48 @@
-import React from 'react';
+import React, { useRef } from 'react';
 import styles from './AddItem.module.scss';
+import useAddTodo from '../../hooks/useAddTodo';
+import {
+  TodoImportance,
+  TodoImportanceValues,
+  TodoToCreate,
+} from '../../types';
+import { capitalize } from '../../helpers';
 
 const AddItem = () => {
+  const { mutate: addTodo } = useAddTodo(resetForm);
+  const addFormRef = useRef<HTMLFormElement>(null);
+
+  function resetForm()  {
+    addFormRef?.current?.reset();
+  }
+
   const handleFormSubmit = (e: React.SyntheticEvent) => {
     e.preventDefault();
     const target = e.target as HTMLFormElement;
-    const objectOfFields = Object.fromEntries(new FormData(target));
+    const formFields = Object.fromEntries(new FormData(target)) as {[k: string]: string} ;
+    const newTodo: TodoToCreate = {
+      title: formFields.todoTitle,
+      importance: formFields.todoImportance as TodoImportance,
+    }
+    addTodo(newTodo);
   }
 
   return (
     <div className={`container ${styles.pageWrapper}`}>
       <h1 className={styles.pageTitle}>Add new todo</h1>
-      <form onSubmit={handleFormSubmit} className={styles.addForm}>
+      <form onSubmit={handleFormSubmit} className={styles.addForm} ref={addFormRef}>
         <div className={styles.formGroup}>
           <label htmlFor="todoTitle">Title:</label>
           <input type="text" id="todoTitle" name="todoTitle" minLength={2} maxLength={130} required />
         </div>
         <div className={styles.formGroup}>
           <label htmlFor="todoImportance">Importance:</label>
           <select id="todoImportance" name="todoImportance">
-            {/*todo: use importanceArr.map to generate options when PR https://github.com/42-alex/todolist/pull/13 will be merged */}
-            <option>Ordinary</option>
-            <option>Important</option>
-            <option>Urgent</option>
+            { TodoImportanceValues.map(importanceTitle =>
+              <option key={importanceTitle} value={importanceTitle}>
+                {capitalize(importanceTitle)}
```

Comment:
I see you want to avoid repetitions. But it's more flexible to keep value and title independent. Initially it might be "almost the same":

```
{
 ordinary: "Ordinary",
 important: "Important"
 urgent: "Urgent"
}
```

but later you may want to change label text _without_ ability to patch existing data in DB. Or to have multiple languages supported for label. This way much better to keep labels and values independent right from the start.

Reply:
@skyboyer
fixed.